src/ethereum/mod.rs

```diff
@@ -134,7 +133,7 @@ impl Ethereum {
         // Get all events for that block from the node and add them to the block struct.
         let web3_clone = self.web3.clone();
         blocks.and_then(move |mut block| {
-            let block_number: u64 = block.number.into();
+            let block_number: u64 = block.number.low_u64();
```

Comment:
Why `low_u64` and not `as_u64`? 
Converting U128 to u64 will not cause an overflow?

Reply:
Good point. I did not find a better solution. The block number expects a `u64` as argument. No idea why 🤦‍♂️And of course a `U128` cannot be safely cast to a `u64`, because it is twice as long.
`u64`'s max value `18446744073709551615` is probably sufficient for the block number. So that would be the `U128`'s *lower* of the two 64 bit unsigned integers.

I will add a comment to the code, thanks for bringing this up 👍